L1Trigger/L1TGEM/plugins/GEMPadDigiProducer.cc

```diff
@@ -96,7 +104,7 @@ void GEMPadDigiProducer::buildPads(const GEMDigiCollection& det_digis, GEMPadDig
 
     // fill the output collections
     for (const auto& d : proto_pads) {
-      GEMPadDigi pad_digi(d.first, d.second);
+      GEMPadDigi pad_digi(d.first, d.second, enumStation(p->id().station()));
```

Comment:
as a future development item, it could be nice to use the station enum everywhere, including in the DetId

Reply:
Do I understand correctly that you suggest to move the `Station` class enum to `GEMDetId`? I suppose it wouldn't be a big deal to include `GEMDetId` in the `GEMPadDigi` and similar objects.

Reply:
alternatively, a separate header like https://github.com/cms-sw/cmssw/blob/master/DataFormats/MuonDetId/interface/MuonSubdetId.h could be introduced

Reply:
E.g. something like this?

```
#ifndef DataFormats_MuonDetId_GEMSubDetId_h
#define DataFormats_MuonDetId_GEMSubDetId_h

/** \class GEMSubDetId
 *
 */

class  GEMSubDetId {
public:
   enum class Station { ME0 = 0, GE11 = 1, GE21 = 2 };
   static Station station(uint16_t st) const { 
     Station returnValue = Station::GE11;
     if (st == 0) {
       returnValue = Station::ME0;
     }
     else if (st == 2) {
       returnValue = Station::GE21;
     }
     return returnValue;
    };
};

#endif
```

Reply:
I was just thinking of a short header file to host the enum class definition, but having the converter function there as well could be a nice touch

Reply:
Cool. Once the results from the tests are in I can make the change in this PR to clean it up nice.

Reply:
I have proposal here: https://github.com/cms-sw/cmssw/compare/master...dildick:CMSSW_11_2_X-GEM-valid-clusters?expand=1, but this is causing memory issues in WF 24834.0. I'm looking into it

```
----- Begin Fatal Exception 17-Jun-2020 15:40:36 CDT-----------------------
An exception of category 'BadAlloc' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 2 stream: 0
   [1] Running path 'HLTAnalyzerEndpath'
   [2] Prefetching for module L1TRawToDigi/'hltGtStage2Digis'
   [3] Prefetching for module RawDataCollectorByLabel/'rawDataCollector'
   [4] Prefetching for module CSCDigiToRawModule/'cscpacker'
   [5] Calling method for module CSCTriggerPrimitivesProducer/'simCscTriggerPrimitiveDigis'
Exception Message:
A std::bad_alloc exception was thrown.
The job has probably exhausted the virtual memory available to the process.
----- End Fatal Exception -------------------------------------------------
```